chapters/common_pitfalls.md

```diff
@@ -0,0 +1,44 @@
+# Common Pitfalls for new Vulkan Developers
+
+This is a short list of common assumptions and traps developers new to Vulkan can make.  
+
+### Validation Layers
+
+During development, ensure that the Validation Layers are enabled. They are an invaluable tool for catching mistakes while using the Vulkan API. Parameter checking, object lifetimes, and threading violations all are part of the provided error checks. A way to reassure that they are enabled is to verify if the text "Debug Messenger Added" is in the output stream. More info can be found in the [Vulkan SDK](https://vulkan.lunarg.com/doc/sdk/latest/windows/layer_configuration.html) layer documentation. It is first on the list because it is the first line of defense and should help reduce the time spent discovering where the mistake was made.
+
+### Vulkan is a box of tools
+
+In Vulkan, there are often multiple solutions to a problem. Consider passing data to a shader, one can use uniform buffers, storage buffers, vertex attributes, instance buffer, push constants, and more. Are all valid methods each with their own benefits and drawbacks. There is rarely a "perfect" solution to any problem. Therefore, use the tools available and create a good solution instead of trying to find a perfect solution.
+
+### Recording command buffers  
+
+A common assumption when first learning about command buffers is that reusing them is paramount and that re-recording them every frame is costly. While it may appear counterintuitive, there is often a greater cost in reuse than simply recording new command buffers each frame. The main cost is the additional complexity for managing dynamic objects, frustum culling, and many other graphical effects. Many of the approaches used to reduce command buffer recording add non trivial state management or undue complexity to the rendering architecture. Therefore, especially in simpler situations, the cost of re-recording every frame is reasonable. Developers still concerned can and should profile command buffer recording to make better informed decisions about the best approach for their application.
```

Comment:
I think it would be better to pretty much rule out command buffer reuse altogether - more detailed chapters could explain the few cases where they are actually useful. Their utility is pretty low in practice - I think the main real world use cases are things like pre-canned layout transitions to stitch command buffers together, or dedicated post-processes.

I'd maybe reword this to start with something like:

> Many early Vulkan tutorials and documents recommend command buffer reuse - but outside of toy examples, this is largely unnecessary and puts significant additional burden on the developer. The gains you might see from pre-recorded command buffers is typically vanishingly small due to the already low overhead of Vulkan. The main cost is the additional complexity for managing...